File: server/src/main/java/by/brawl/ws/holder/GameSession.java
package by.brawl.ws.holder;

import by.brawl.util.Exceptions;
import by.brawl.ws.dto.JsonDto;
import by.brawl.ws.dto.MessageDto;
import org.codehaus.jackson.map.ObjectMapper;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.socket.TextMessage;
import org.springframework.web.socket.WebSocketSession;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public class GameSession {

	private static final Logger LOG = LoggerFactory.getLogger(GameSession.class);

	private String id;
	private WebSocketSession session;
	private BattlefieldHolder battlefieldHolder;

	public GameSession(WebSocketSession session) {
		this.session = session;
		id = session.getPrincipal().getName();
	}

	public void sendInfoMessage(String text) {
		sendDto(new MessageDto(text));
	}

	public void sendDto(JsonDto dto) {
		try {
			session.sendMessage(new TextMessage(dto.asJson()));
		} catch (IOException e) {
			Exceptions.logError(LOG, e, "Web socket message sending threw error");
		}
	}

	public void sendKeyValue(String key, Object value) {
		try {
			Map<String, Object> map = new HashMap<>();
			map.put(key, value);
			session.sendMessage(new TextMessage(new ObjectMapper().writeValueAsString(map)));
		} catch (IOException e) {
			Exceptions.logError(LOG, e, "Web socket message sending threw error");
		}
	}

	public String getId() {
		return id;
	}

	public BattlefieldHolder getBattlefieldHolder() {
		return battlefieldHolder;
	}

	public void setBattlefieldHolder(BattlefieldHolder battlefieldHolder) {
		this.battlefieldHolder = battlefieldHolder;
	}
}

File: server/src/main/java/by/brawl/ws/holder/gamesession/GameSession.kt
package by.brawl.ws.holder.gamesession

import by.brawl.ws.dto.JsonDto
import by.brawl.ws.dto.MessageDto
import by.brawl.ws.holder.RoomHolder
import org.codehaus.jackson.map.ObjectMapper
import org.springframework.web.socket.TextMessage
import org.springframework.web.socket.WebSocketSession
import java.util.*

class GameSession(private val session: WebSocketSession) {

    val username: String = session.principal.name
    lateinit var roomHolder: RoomHolder

    fun sendInfoMessage(text: String) {
        sendDto(MessageDto(text))
    }

    fun sendDto(dto: JsonDto) {
        session.sendMessage(TextMessage(dto.asJson()))
    }

    fun sendKeyValue(key: String, value: Any) {
        val map = HashMap<String, Any>()
        map.put(key, value)
        session.sendMessage(TextMessage(ObjectMapper().writeValueAsString(map)))
    }

}